controllers/xsite.go

```diff
@@ -43,18 +39,64 @@ func (r *infinispanRequest) applyLabelsToCoordinatorsPod(podList *corev1.PodList
 					crossSiteViewFormed.Message = fmt.Sprintf("Error: %s", err.Error())
 				}
 				return crossSiteViewFormed, nil
-			} else {
-				if ok && lab == strconv.FormatBool(ok) {
-					// If present leave the label but false the value
-					if ok {
-						item.Labels[consts.CoordinatorPodLabel] = strconv.FormatBool(cacheManagerInfo.Coordinator)
-						if err = r.Client.Update(r.ctx, &item); err != nil {
-							return nil, err
-						}
-					}
-				}
 			}
 		}
 	}
 	return &ispnv1.InfinispanCondition{Type: ispnv1.ConditionCrossSiteViewFormed, Status: metav1.ConditionFalse, Message: "Coordinator not ready"}, nil
 }
+
+// GetGossipRouterDeployment returns the deployment for the Gossip Router pod
+func (r *infinispanRequest) GetGossipRouterDeployment(m *infinispanv1.Infinispan) *appsv1.Deployment {
+	name := m.GetGossipRouterDeploymentName()
+	lsTunnel := GossipRouterPodLabels(m.Name)
+	replicas := int32(1)
+
+	// if the user configures 0 replicas, shutdown the gossip router pod too.
+	if m.Spec.Replicas <= 0 {
+		replicas = 0
+	}
+
+	deployment := &appsv1.Deployment{
```

Comment:
@pruivo what do you think about `gossiprouter` sidecar, not separate deployment? Sidecar shares network/storage with the Infinispan container, but number of the sidecars will be the same as cluster size.

Reply:
There is no advantage of having more than one GossipRouter. I would have to tag a single pod to be used with the exposed service and that pod will be hit with all the traffic from both sites (and it will negate the multi site master advantage).
If that's ok for you, I would rather a the gossip router in its own pod.

Reply:
Yes, sure. The sidecar additional value is a tightly coupling within the server Pods. If it's not so important, we could live it as is.